src/org/parosproxy/paros/core/scanner/VariantAbstractQuery.java

```diff
@@ -123,6 +124,10 @@ protected void setParameters(int type, List<org.zaproxy.zap.model.NameValuePair>
             listParam.add(new NameValuePair(type, nonNullString(parameter.getName()), nonNullString(parameter.getValue()), i));
             i++;
         }
+        if (i == 0) {
+        	// No query params, lets add one just to make sure
+            listParam.add(new NameValuePair(type, "query", "query", i));

```

Comment:
I'm OK with both ways. We should document the behaviour in the help pages, though.